### PR TITLE
Comic Images Storage

### LINKED ISSUE #133: Add a route to retrieve images
I'll figure out a URL pattern and mention it in Slack.

### LINKED ISSUE #167: Store and retrieve image data through an Image entity
Here's an idea; have `Comic`s store not the actual image data, but just a URL.  This URL can refer to one of two places; either a DeviantArt image, or an image stored ourselves through Mixxy (or even _another_ provider if the need arises down the line).  The actual image data would be stored in a new `Image` model (though maybe not that name, since I bet we'll be using the App Engine API's `Image` class in places; maybe `ImageData`?).

I said in person that we probably wouldn't need a new controller, but a few seconds further of reflection and I actually think it would be a good idea.  `ImageController` can be lower-level; `ComicController` will just load a comic regardless of where it's actually stored, and `ImageController` will actually save and load `ImageData` entities, which are pretty much just binary blobs.

[It looks like Objectify automatically treats `byte[]`s as binary blobs](https://github.com/objectify/objectify/wiki/Entities#byte), which is nice.

**However.**  It turns out that GAE *does* have a limit on Data Store space for free plans.  This limit is 1GB.  We might as well use Google Cloud Storage or Blobstore (looks like the former is suggested).  I was hesitant to use one of these because I don't want to spend money, but if we're going to continue this after the semester ends then we could actually put time into monetizing Mixxy (at least enough to cover storage costs).

### LINKED ISSUE #216: Allow user to upload images
Yes, Muro lets you import a layer, but we don't want to force Muro usage.

### PR DESCRIPTION
**Pull request by Bsabz**
Fix #167 
Fix #133 
Fix #216

### CODE REVIEW

File: src/main/java/www/purple/mixxy/models/ImageData.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	
+	@Id
+	public Long id;
+	
+	@Index
+	public Long comicId;
+	
+	public byte[] imageBlob;
+	
+	public Date createdAt;
```
Comment (JesseTG) [inline src/main/java/www/purple/mixxy/models/ImageData.java]: Do we really care about when an image was created or updated?

Comment (Bsabz) [inline src/main/java/www/purple/mixxy/models/ImageData.java]: Yes, for development and debugging purposes.

Comment (JesseTG) [inline src/main/java/www/purple/mixxy/models/ImageData.java]: Ah, okay.

**Comment by JesseTG** [comment]
So far, so good.  Now let's put together an `ImageController`.  After class tonight, we can work on this 'til the cows come home.

**Comment by Bsabz** [comment]
@JesseTG Take a look at this latest commit. Not exactly sure on how to write the Image Controller with our current structure of the Comic Controller. Plus how do we define the routes?

**Comment by JesseTG** [comment]
As I see it, `ImageData` should literally *not care* about the notion of a `Comic`.  `ImageDao` should, but only to the extent of "I have this `Comic`, get me the actual image."  `Given a URL (one that might not even refer to a `Comic`), get an `ImageData`.  Does it come from Mixxy or from DeviantArt?  I dunno.  I don't care.  That's `ImageDao`'s job to figure out, and it's `ImageController`'s job to spit out the results to the client.

As for how the routes should work?  Good question, we'll talk after class.

File: src/main/java/www/purple/mixxy/conf/ObjectifyProvider.java [hunk 2]
```diff
@@ -35,27 +40,75 @@ public Objectify get() {
...
+
+        return outputStream.toByteArray();
+    }
+    
     public static void setup() {
         Objectify ofy = ObjectifyService.ofy();
         User user = ofy.load().type(User.class).first().now();
 
         User bob = new User("BobTheBuilder", "Bob", "Smith", "male", "bob@gmail.com", "url", "en", "123", "google");
-        bob.id = (long) -32;
```
Comment (JesseTG) [inline src/main/java/www/purple/mixxy/conf/ObjectifyProvider.java]: Any particular reason you made these positive?

**Pull request merged**